include q metrics with different return df types

dominikburri opened the pull request:
Fixes #325 

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated corresponding READMEs (if applicable)
- [x] My code follows the templates/style guidelines of the repository
- [x] No parameters or file names are hardcoded
- [x] Results, logs or other output is not commited to the repository
- [x] I have tested my code

ninsch3000 reviewed (approved):
Not tested but LGTM

faricazjj reviewed (dismissed):
LGTM :D

faricazjj commented on summary_workflows/quantification/specification/Q_benchmark_specification.md:
```diff
@@ -190,15 +190,16 @@ The following tables list the metric names, value types and units, and a descrip
 | `Expression_non-matched-PAS_100nt` | `float` | N/A | Total expression of non-matched sites quantified by benchmarked tool for 100 nt distance threshold (window) |
-| `Correlation_coefficient_Pearson_relative_100nt` | `float` | N/A | Correlation between relative PAS usage calculated from RNAseq-based PAS quantification and orthogonal 3'end seq data for 100 nt distance threshold (window) |
+| `Correlation_coefficient_Pearson_relative_with_unmatched_GT_and_PD_100nt` | `float` | N/A | Correlation between relative PAS usage calculated from RNAseq-based PAS quantification and orthogonal 3'end seq data for 100 nt distance threshold (window) |
 
```
Are we possibly missing the description of without unmatched in this table?

dominikburri replied:
Mh in this table in the specification only metrics are described that appear in the plots. The note below mentions that all three versions are computed and available in the `assessment_dataset.json` and in the `consolidated_result.json`.
My intend was to keep the computations as they were but include the additional ones in the json files. Once we settle on the metric to plot and to use, we can adjust it here in the specification. 
Do you agree?

faricazjj replied:
Ah i see! That makes sense! I thought it was a list of the metrics we have now hehe if that's the case then I was thinking maybe we could add somewhere a consolidated list of the possible metric calculation options and their descriptions, but I guess that should be a separate PR :D

faricazjj reviewed (commented):
Thank you @dominikburri ! :D I just have one question, everything else looks good!

faricazjj reviewed (approved):
LGTM thank you @dominikburri !!

ninsch3000 commented on summary_workflows/quantification/specification/example_files/assessment_out.json:
```diff
@@ -1,46 +1,46 @@
 [
```
I'm afraid this example file is not correct anymore if you loop over `windows` AND `return_df_types` in `compute_metrics.py`. However, it would be quite an overhead to specify all metrics combinations. So maybe you can add some "..." and specify at least each return type once. Or instead of writing the exact id/name, write it with [METRIC_NAME] and [RETURN_TYPE] or sth like that.

dominikburri replied:
I cannot add comments in json files. So I can add all metrics that are also in the example, in total 27.

ninsch3000 commented on summary_workflows/quantification/specification/Q_benchmark_specification.md:
```diff
@@ -190,15 +190,16 @@ The following tables list the metric names, value types and units, and a descrip
+| `Correlation_coefficient_Pearson_relative_with_unmatched_GT_and_PD_100nt` | `float` | N/A | Correlation between relative PAS usage calculated from RNAseq-based PAS quantification and orthogonal 3'end seq data for 100 nt distance threshold (window) |
 
 > The correlation metrics above are also computed for Spearman correlation, but currently not used in the plots. The metric ids are adjusted accordingly.
+> The correlation metrics above are besides *with_unmatched_GT* and *with_unmatched_GT_and_PD* for relative correlation also computed for *with_unmatched_GT_and_PD*, *without_unmatched* and *with_unmatched_GT*. That is, all three variants are computed, but only the ones above are plotted. The metric ids are adjusted accordingly.
```
Could you rephrase that? E.g.   
"The above mentioned correlation metrics are actually calculated for three different matching types - `with_unmatched_GT`, `with_unmatched_GT_and_PD` and `without_unmatched`, and for Pearson's as well as Spearman's r. All these variants will appear in the `assessment_out.json` and `consolidated_results.json`. However, the `aggregation_out.json` will only contain the metrics explicitly specified for visualization in `aggregation_template.json`"

ninsch3000 commented on summary_workflows/quantification/quantification_dockers/q_metrics/compute_metrics.py:
```diff
@@ -52,53 +54,55 @@ def compute_metrics(participant_input, gold_standards_dir, challenge_ids, partic
+                # METRIC: Matched sites
+                ########################
+                # metric on the number of matched sites
+                match_with_gt_run = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard, 
```
With the refactoring you introduced (returning the df here and the passing it to the Pearson and Spearman functions), this call should actually go above "METRIC: Matched sites". The return of `match_with_gt` is used in the three subsequent metric calculations

ninsch3000 commented on summary_workflows/quantification/quantification_dockers/q_metrics/compute_metrics.py:
```diff
@@ -52,53 +54,55 @@ def compute_metrics(participant_input, gold_standards_dir, challenge_ids, partic
+                ####################
+                # Only calculate metric for first window size.
+                if window == windows[0]:
+                    match_with_gt_and_pd = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard,
```
the variable names `match_with_gt_and_pd`, as well as `merged_bed_df_wpd` are misleading after the refactoring you did, because now this df is calculated for all three return types. Actually, couldn't you therefore even use the `merged_bed_df` from above? As far as I understand the code there is no need to re-calculate it.

dominikburri replied:
good catch! Will change

ninsch3000 reviewed (changes requested):
Thanks a lot @dominikburri . I think it would be nice to further clean up this PR, see my individual comments.    
Moreover, should we maybe create a new issue to make shorter metric names? Like `Pearson_r` instead of `Correlation_coefficient_Pearson_...`.   
After you address my comments I'd still want to test the scripts with the small example file to check the calculations. Just let me know when it's ready :pray:

dominikburri commented:
> Thanks a lot @dominikburri . I think it would be nice to further clean up this PR, see my individual comments.
> Moreover, should we maybe create a new issue to make shorter metric names? Like `Pearson_r` instead of `Correlation_coefficient_Pearson_...`.
> After you address my comments I'd still want to test the scripts with the small example file to check the calculations. Just let me know when it's ready 🙏

Thanks @ninsch3000, I addressed your comments. 
As for the metric names, I think a new PR together with @faricazjj's suggestion of more detailed description should be in a separate PR.

faricazjj reviewed (approved):
LGTM thanks @dominikburri ! :D
